BDW/metrics/perf/broadwell_metrics_perf.json

```diff
@@ -1,586 +1,576 @@
 [
     {
-        "BriefDescription": "This category represents fraction of slots where the processor's Frontend undersupplies its Backend. Frontend denotes the first part of the processor core responsible to fetch operations that are executed later on by the Backend part. Within the Frontend; a branch predictor predicts the next address to fetch; cache-lines are fetched from the memory subsystem; parsed into instructions; and lastly decoded into micro-operations (uops). Ideally the Frontend can issue Pipeline_Width uops every cycle to the Backend. Frontend Bound denotes unutilized issue-slots when there is no Backend stall; i.e. bubbles where Frontend delivered no uops while Backend could have accepted them. For example; stalls due to instruction-cache misses would be categorized under Frontend Bound.",
+        "BriefDescription": "This category represents fraction of slots where the processor's Frontend undersupplies its Backend.",
```

Comment:
The existing perf json lacks a period on the end of the descriptions:
https://git.kernel.org/pub/scm/linux/kernel/git/perf/perf-tools-next.git/tree/tools/perf/pmu-events/arch/x86/broadwell/bdw-metrics.json?h=perf-tools-next#n330
```
        "BriefDescription": "This category represents fraction of slots where the processor's Frontend undersupplies its Backend",
```
So we'll get diffs because of the periods here. The perf tool will insert periods on the end of descriptions in perf list, so maybe we should lose them here or update the perf json importing code to drop them. We'll get all descriptions diffing in the LKML upload because of these periods though.

Reply:
I checked out your branch from #241 and rebased it on top of main to review the output. It looks like the description substitution `sdesc = re.sub(r'(?<!i\.e)\. .*', '', desc)` drops trailing periods when making a short description.

```
Creating event json for BDW in /home/ed/perfmon/public/perfmon/scripts/out-metric-descriptions-2/broadwell
Description for tma_frontend_bound on BDW is:
 desc="This category represents fraction of slots where the processor's Frontend undersupplies its Backend. Frontend denotes the first part of the processor core responsible to fetch operations that are executed later on by the Backend part. Within the Frontend; a branch predictor predicts the next address to fetch; cache-lines are fetched from the memory subsystem; parsed into instructions; and lastly decoded into micro-operations (uops). Ideally the Frontend can issue Pipeline_Width uops every cycle to the Backend. Frontend Bound denotes unutilized issue-slots when there is no Backend stall; i.e. bubbles where Frontend delivered no uops while Backend could have accepted them. For example; stalls due to instruction-cache misses would be categorized under Frontend Bound."
sdesc="This category represents fraction of slots where the processor's Frontend undersupplies its Backend"
```

In the above example log `sdesc` and then `formula['BriefDescription']` match https://git.kernel.org/pub/scm/linux/kernel/git/perf/perf-tools-next.git/tree/tools/perf/pmu-events/arch/x86/broadwell/bdw-metrics.json?h=perf-tools-next#n330